proxy/http/HttpTransact.cc

```diff
@@ -1856,12 +1852,9 @@ HttpTransact::ReDNSRoundRobin(State *s)
 void
 HttpTransact::OSDNSLookup(State *s)
 {
-  static const int max_dns_lookups = 3;
-
   ink_assert(s->dns_info.looking_up == ORIGIN_SERVER);
 
-  TxnDebug("http_trans", "[HttpTransact::OSDNSLookup] This was attempt %d", s->dns_info.attempts);
```

Comment:
I'm not clear on the `DNS_ATTEMPTS_EXHAUSTED` and `max_dns_lookups` removal below. 
Do these changes make it impossible for `OSDNSLookup` to ever be called more than once in a Transaction?

If not, is this debug message useful to people?
If so, I see asserts below, how do we now make sure we don't somehow have a loop?
Also, with the removal of the "expanded name" stuff, why would `OSDNSLookup` ever need to be called more than once for the same transaction?

Reply:
I removed DNS_ATTEMPTS_EXHAUSTED because the current try_to_expand_host_name() never returns that.  It only returns EXPANSION_NOT_ALLOWED in the non-parent selection case or RETRY_EXPANDED_NAME in the parent selection case.  So got rid of the extra cases and inlined the contents of try_to_expand host name.

With the most of the expandomatic logic gone, only the parent select case exercises this path, and it will only exercise it once to check the IP address of the parent machine rather than the parent path directly (here is where my knowledge of why the parent select logic flips over to OSDNSLookup becomes shakey).

The underlying DNS lookup logic in iocore/DNS will still do multiple DNS requests. 

The origin server looking up logic will pass through OSDNSLookup multiple times for a transaction if the connection for an IP address fails . Then OSDNSLookup will be triggered again and either the same address is selected again or if it failed too many times, the logic will move onto the next address (assuming DNS returns multiple addresses for the name)

Reply:
I was mostly worried about debugging loops. I said transaction, I meant host. It sounds like it shouldn't get called for the same host repeatedly, and if it is, if there is a loop, there should be other debug messages. Besides which, I think parent selection is the right place to log "this is lookup N" not here.